commands/http/client.go

```diff
@@ -103,46 +96,27 @@ func (c *client) Send(req cmds.Request) (cmds.Response, error) {
 	version := config.CurrentVersionNumber
 	httpReq.Header.Set(uaHeader, fmt.Sprintf("/go-ipfs/%s/", version))
 
-	ec := make(chan error, 1)
-	rc := make(chan cmds.Response, 1)
-	dc := req.Context().Done()
+	httpReq.Cancel = req.Context().Done()
```

Comment:
can we have a test here to make sure this works as we expect it to? like have some request hang and have a context expire + be cancelled out? (doesnt have to be sharness, could be golang test)